rebasehelper/helpers/process_helper.py

```diff
@@ -0,0 +1,205 @@
+#
+# This tool helps you to rebase package to the latest version
+# Copyright (C) 2013-2014 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Authors: Petr Hracek <phracek@redhat.com>
+#          Tomas Hozza <thozza@redhat.com>
+
+import os
+import subprocess
+import tempfile
+
+import six
+
+from rebasehelper.constants import defenc
+from rebasehelper.logger import logger
+
+
+class ProcessHelper(object):
+
+    """Class for executing subprocess."""
+
+    DEV_NULL = os.devnull
+
+    @staticmethod
+    def run_subprocess(cmd, input_file=None, output_file=None):
+        """Runs the passed command as a subprocess.
+
+        Args:
+            cmd (list): Command with arguments to be run.
+            input_file (str): File to read the input from. If None, read from STDIN.
+            output_file (str): File to write the output of the command to. If None, write to STDOUT.
+
+        Returns:
+            int: Exit code of the process.
+
+        """
+        return ProcessHelper.run_subprocess_cwd(cmd, input_file=input_file, output_file=output_file)
+
+    @staticmethod
+    def run_subprocess_cwd(cmd, cwd=None, input_file=None, output_file=None, shell=False):
+        """Runs the passed command as a subprocess in a different working directory.
+
+        Args:
+            cmd (list): Command with arguments to be run.
+            cwd (str): Working directory for the command.
+            input_file (str): File to read the input from. If None, read from STDIN.
+            output_file (str): File to write the output of the command to. If None, write to STDOUT.
+            shell (bool): Whether to run the command as shell command.
+
+        Returns:
+            int: Exit code of the process.
+
+        """
+        return ProcessHelper.run_subprocess_cwd_env(cmd,
+                                                    cwd=cwd,
+                                                    input_file=input_file,
+                                                    output_file=output_file,
+                                                    shell=shell)
+
+    @staticmethod
+    def run_subprocess_env(cmd, env=None, input_file=None, output_file=None, shell=False):
+        """Runs the passed command as a subprocess with possibly changed ENVIRONMENT VARIABLES.
```

Comment:
And this I'd change to `Runs the specified command in a subprocess with a redefined environment.`